environmentcleanup/src/main/java/com/axway/ats/environment/database/MssqlEnvironmentHandler.java

```diff
@@ -110,10 +111,15 @@ protected void writeTableToFile(
                                      DbRecordValuesList[] records,
                                      FileWriter fileWriter ) throws IOException, ParseException {
 
-        if( this.includeDeleteStatements ) {
-            fileWriter.write( "DELETE FROM " + table.getTableName() + ";" + EOL_MARKER
-                              + AtsSystemProperties.SYSTEM_LINE_SEPARATOR );
+        if( this.includeDeleteStatements && !this.deleteStatementsInserted) {
```

Comment:
This check and loop for delete statements could be extracted and used in common method in abstract class. The additional flag to be checked could be evaluated before invoking this method.